src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -248,21 +249,31 @@ public SyntaxTriviaList InsertRange(int index, IEnumerable<SyntaxTrivia> trivia)
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
 
-            var items = trivia.ToList();
-            if (items.Count == 0)
+            if (trivia == null)
             {
-                return this;
+                throw new ArgumentNullException(nameof(trivia));
             }
 
-            var list = this.ToList();
-            list.InsertRange(index, items);
+            var builder = SyntaxTriviaListBuilder.Create();
+
+            var current = 0;
+            foreach (var thisTrivia in this)
```

Comment:
this simply seemed like the easiest way to code this up, while also removing several allocations (allocation of two lists, an enumerable, an enumerator, and a lambda).  The 'for' approach seemed less clear to me for some reason :)